mono/sgen/sgen-pointer-queue.c

```diff
@@ -119,7 +119,7 @@ sgen_pointer_queue_sort_uniq (SgenPointerQueue *queue)
 		start++;
 	};
 	queue->next_slot = start - queue->data;
-	SGEN_LOG (5, "Pointer queue reduced to size: %lu", queue->next_slot);
+	SGEN_LOG (5, "Pointer queue reduced to size: %zu", queue->next_slot);
```

Comment:
%z is not portable to msvc runtime.

Reply:
This is portable for msvc versions we support VS2015+ and its VS2015 triggering warning around use of %lu for size_t so fine there, but it still has issues on mingw. Since original is %lu it should be OK to type cast to unsigned long to not bring up additional warnings on mingw.